### ISSUE TITLE
tasty-hunit doesn't import hunit

### ISSUE THREAD

**Issue by ocramz** [feuerbach/tasty#230]
I just noticed there's code copied from `hunit`, what about importing it and re-exporting the relevant things instead?

**Comment by feuerbach** [comment]
This is a feature, not a bug. https://github.com/feuerbach/tasty/blob/master/hunit/CHANGELOG.md#version-09

Is there a specific problem that you ran into?

**Comment by ocramz** [comment]
I noticed this in a test, when catching a `HUnitFailure`, since its number of parameters changed in a recent version. I thought that, given its name `tasty-hunit` would import both, which would make it easier to track API changes.

**Comment by feuerbach** [comment]
The name is a legacy of the early versions of tasty-hunit and of test-framework-hunit, which did depend on HUnit. I'll add a note to clarify this. The parts of HUnit that were used are rather trivial, and a big part of HUnit (dealing with running tests) is useless and incompatible with tasty, so it's hard to justify the burden of a dependency.

**Comment by gromakovsky** [comment]
I would say ran into a specific problem. I have some tests that are `Assertion`s and use functions from `HUnit` itself. I was running them using something different from `tasty` and then I start using `tasty` and `tasty-discover`. And I was very surprised when I noticed that if some test fails I don't get pretty output, but instead get an uncaught exception. The reason is that tests throw `HUnitFailure` from `HUnit` while `tasty-hunit` catches another `HUnitFailure`.

If `HUnit` functions are used directly in my code I can fix it by replacing `HUnit` import with `Test.Tasty.HUnit`. But sometimes there may be a helper library function for testing which depends on `HUnit`. And I can't use these functions together with `tasty-hunit`.

So the main problem I have is that it breaks interaction with existing code which uses `HUnit` and it's quite hard to notice (I noticed it only when some test failed).

Doesn't it make sense to add a more generalized version of `testCase` (and other functions) which allows one to decide which `HUnitFailure` to catch (without depending on `HUnit`)? Parameterize it by `e` (where `e` is `Test.Tasty.HUnit.HUnitFailure` in version) and require a function which converts `e` to something that is needed in the error handler.
